File: src/common/util/loaders/AudioBuffer.java
package common.util.loaders;

import javafx.scene.media.Media;
import javafx.scene.media.MediaException;

import java.net.URISyntaxException;
import java.util.HashMap;

public final class AudioBuffer {

    private static final HashMap<String, Media> fileToMedia = new HashMap<>();

    private static void addFile( final String filepath ) throws MediaException {
        try {
            fileToMedia.put(
                    filepath,
                    newMedia( filepath )
                           );
        }
        catch ( NullPointerException npe ) {
            npe.printStackTrace();
        }
    }

    private static Media newMedia( final String fileName ) throws MediaException {
        try {
            return new Media( AudioBuffer.class.getResource( fileName ).toURI().toString() );
        }
        catch ( URISyntaxException e ) {
            e.printStackTrace();
        }
        return null;
    }

    private static boolean contains( final String fileName ) {
        return fileToMedia.containsKey( fileName );
    }

    public static Media loadMedia( final String fileName ) throws MediaException {
        if ( !AudioBuffer.contains( fileName ) ) {
            AudioBuffer.addFile( fileName );
        }
        return fileToMedia.get( fileName );
    }

}

File: Common/Sound/src/common/sound/AudioBuffer.kt
package common.sound

import javafx.scene.media.Media
import javafx.scene.media.MediaException
import java.io.File
import java.io.FileNotFoundException
import java.util.*

object AudioBuffer {
    private val fileToMedia = HashMap<String, Media?>()

    @Throws(MediaException::class)
    private fun addFile(filepath: String) {
        try {
            fileToMedia[filepath] = newMedia(filepath)
        } catch (npe: NullPointerException) {
            npe.printStackTrace()
        } catch (npe: FileNotFoundException) {
            npe.printStackTrace()
        }
    }

    @Throws(MediaException::class,
            FileNotFoundException::class)
    private fun newMedia(fileName: String): Media {
        val url = AudioBuffer::class.java.getResource(fileName)
                  ?: throw FileNotFoundException(
                          fileName)
        val soundFile = File(url.path)
        if (soundFile.exists()) {
            return Media(soundFile.toURI().toString())
        }
        throw FileNotFoundException(fileName)
    }

    private operator fun contains(fileName: String): Boolean {
        return fileToMedia.containsKey(fileName)
    }

    @Throws(MediaException::class)
    fun loadMedia(fileName: String): Media? {
        if (!contains(fileName)) {
            addFile(fileName)
        }
        return fileToMedia[fileName]
    }
}
